odoo/addons/base/views/ir_qweb_widget_templates.xml

```diff
@@ -28,12 +28,18 @@
                 <i t-if="not options.get('no_marker')" class="fa fa-map-marker fa-fw" role="img" aria-label="Address" title="Address"/>
                 <span class="w-100 lh-sm text-truncate" itemprop="streetAddress" t-esc="address"/>
             </div>
-            <div t-if="city and 'city' in fields" t-attf-class="d-flex align-items-baseline gap-1">
+            <t t-set="has_city" t-value="city and 'city' in fields"/>
+            <t t-set="has_state" t-value="object.state_id and object.country_id.state_required"/>
+            <t t-set="has_country" t-value="country_id and 'country_id' in fields "/>
+            <t t-set="location" t-value="[]"/>
+            <t t-if="has_city" t-set="location" t-value="location + [city]"/>
+            <t t-if="has_state" t-set="location" t-value="location + [object.state_id.code]"/>
```

Comment:
Should probably use a ``[addressRegion](https://schema.org/addressRegion)`` itemprop

Reply:
Hello @tde-banana-odoo :smile: 
Quite confused with this one I investigated a bit and found that we generally use the item prop for the tags that are rendered in dom as here we are just setting the value and using that value to be rendered in just a single span probably we could use `location` as a itemprop there.
But if we need to put the itemprop for all of them probably we need to put them in different tag instead of collecting them in to a single list and joining at the end what currently we are doing may be in that scenario we don't need to compute `has_city, has_country...`
Let me know your opinion or am I missing something.

Reply:
Well idea is to still be compliant with the schema. I don't think stringifying a list and putting everything in addressLocality is the way to go ^^ . We should effectively split. Before we have addressLocality and addressCountry, we should continue using schema bits and add what is requested by the task.